### PR TITLE
Update rke2-hardening-guide based on CIS-1.24 and CIS-1.7 

### PR DESCRIPTION
**Pull request by andypitcher**
This PR updates the [RKE2 hardening guide](https://ranchermanager.docs.rancher.com/pages-for-subheaders/rke2-hardening-guide) with the new requirements added in cis-1.24 and cis-1.7.

Related issue:
- https://github.com/rancher/rancher/issues/42154

### CODE REVIEW

File: docs/pages-for-subheaders/rke1-hardening-guide.md [hunk 1]
```diff
@@ -12,10 +12,11 @@ This hardening guide is intended to be used for RKE clusters and is associated w
 
 | Rancher Version | CIS Benchmark Version | Kubernetes Version           |
 |-----------------|-----------------------|------------------------------|
-| Rancher v2.7    | Benchmark v1.23       | Kubernetes v1.23 up to v1.25 |
+| Rancher v2.7    | Benchmark v1.7       | Kubernetes v1.24 up to v1.25 |
 
 :::note
-At the time of writing, the upstream CIS Kubernetes v1.25 benchmark is not yet available in Rancher. At this time Rancher is using the CIS v1.23 benchmark when scanning Kubernetes v1.25 clusters. Due to that, the CIS checks 5.2.3, 5.2.4, 5.2.5 and 5.2.6 might fail.
+- Since Benchmark v1.24, check id `4.1.7 Ensure that the certificate authorities file permissions are set to 600 or more restrictive (Automated)` might fail, as /etc/kubernetes/ssl/kube-ca.pem is provisioned in 644 by default.
+- Since Benchmark v1.7 (latest), `--protect-kernel-defaults` (check id 4.2.6) parameter is not required anymore, and was replaced.
```
Comment (martyav) [inline docs/pages-for-subheaders/rke1-hardening-guide.md:19]:
```suggestion
- In Benchmark v1.24 and later, check id `4.1.7 Ensure that the certificate authorities file permissions are set to 600 or more restrictive (Automated)` might fail, as `/etc/kubernetes/ssl/kube-ca.pem` is set to 644 by default.
- In Benchmark v1.7, `--protect-kernel-defaults` (check id 4.2.6) parameter isn't required anymore, and was replaced.
```

Similar comment as https://github.com/rancher/rancher-docs/pull/794#discussion_r1302017920, "Should we add what the --protect-kernel-defaults parameter was replaced by? If not, should we remove mention that it was replaced?"

File: docs/pages-for-subheaders/rke2-hardening-guide.md [hunk 2]
```diff
@@ -12,10 +12,11 @@ This hardening guide is intended to be used for RKE2 clusters and is associated
 
 | Rancher Version | CIS Benchmark Version | Kubernetes Version           |
 |-----------------|-----------------------|------------------------------|
-| Rancher v2.7    | Benchmark v1.23       | Kubernetes v1.23 up to v1.25 |
+| Rancher v2.7    | Benchmark v1.7       | Kubernetes v1.24 up to v1.25 |
 
 :::note
-At the time of writing, the upstream CIS Kubernetes v1.25 benchmark is not yet available in Rancher. At this time Rancher is using the CIS v1.23 benchmark when scanning Kubernetes v1.25 clusters. Due to that, the CIS checks 5.2.2, 5.2.3, 5.2.5, 5.2.6, 5.2.7 and 5.2.8 might fail.
+- Since Benchmark v1.24, some check ids might fail due to file permission new requirements (600 instead of 644). Impacted check ids: `1.1.1`, `1.1.3`, `1.1.5`, `1.1.7`, `1.1.13`, `1.1.15`, `4.1.7`, `4.1.9`, `4.1.15`. 
+ - Since Benchmark v1.7 (latest), `--protect-kernel-defaults` (check id 4.2.6) parameter is not required anymore, and was replaced.
```
Comment (martyav) [inline docs/pages-for-subheaders/rke2-hardening-guide.md:19]:
```suggestion
- In Benchmark v1.24 and later, some check ids might fail due to new file permission requirements (600 instead of 644). Impacted check ids: `1.1.1`, `1.1.3`, `1.1.5`, `1.1.7`, `1.1.13`, `1.1.15`, `4.1.7`, `4.1.9`, `4.1.15`. 
 - In Benchmark v1.7, `--protect-kernel-defaults` (check id 4.2.6) parameter is not required anymore, and was replaced.
```

**Review by martyav — CHANGES_REQUESTED** [review]

File: docs/pages-for-subheaders/rke1-hardening-guide.md (hunk 1 above)
Comment (andypitcher) [reply docs/pages-for-subheaders/rke1-hardening-guide.md:19]: @martyav thanks for the review. I think we should remove "and was replaced". In fact, the check id `4.2.6` has been assigned to another check, [here](https://github.com/rancher/security-scan/blob/011ad94bd1a707646e8de880cc1a780f6591b045/package/cfg/rke-cis-1.7-hardened/node.yaml#L283C4-L283C4).

File: docs/pages-for-subheaders/rke2-hardening-guide.md (hunk 2 above)
Comment (andypitcher) [reply docs/pages-for-subheaders/rke2-hardening-guide.md:19]: Suggestion and check ids updated @Priyashetty17 .

File: docs/pages-for-subheaders/rke1-hardening-guide.md [hunk 3]
```diff
@@ -12,10 +12,11 @@ This hardening guide is intended to be used for RKE clusters and is associated w
 
 | Rancher Version | CIS Benchmark Version | Kubernetes Version           |
 |-----------------|-----------------------|------------------------------|
-| Rancher v2.7    | Benchmark v1.23       | Kubernetes v1.23 up to v1.25 |
+| Rancher v2.7    | Benchmark v1.7       | Kubernetes v1.24 up to v1.25 |
```
Comment (andypitcher) [inline docs/pages-for-subheaders/rke1-hardening-guide.md:15]:
```suggestion
| Rancher v2.7    | Benchmark v1.7       | Kubernetes v1.25 |
```

File: docs/pages-for-subheaders/rke2-hardening-guide.md (hunk 2 above)
Comment (martyav) [reply docs/pages-for-subheaders/rke2-hardening-guide.md:19]: @andypitcher did you mean to commit this suggestion?

File: docs/pages-for-subheaders/rke1-hardening-guide.md [hunk 4]
```diff
@@ -12,10 +12,11 @@ This hardening guide is intended to be used for RKE clusters and is associated w
 
 | Rancher Version | CIS Benchmark Version | Kubernetes Version           |
 |-----------------|-----------------------|------------------------------|
-| Rancher v2.7    | Benchmark v1.23       | Kubernetes v1.23 up to v1.25 |
+| Rancher v2.7    | Benchmark v1.7       | Kubernetes v1.25 |
```
Comment (JonCrowther) [inline docs/pages-for-subheaders/rke1-hardening-guide.md:15]: Not sure if this is intentional, but aren't these the same changes from your RKE1 PR? 
https://github.com/rancher/rancher-docs/pull/794/files

I'm ok with the changes being in one PR, but if you'd like to keep them separate I think you should undo this change and leave it to the other PR.

**Comment by JonCrowther** [comment]
You'll need to update the versioned docs as well:
https://github.com/rancher/rancher-docs/blob/main/versioned_docs/version-2.7/pages-for-subheaders/rke2-hardening-guide.md

File: docs/pages-for-subheaders/rke2-hardening-guide.md (hunk 2 above)
Comment (andypitcher) [reply docs/pages-for-subheaders/rke2-hardening-guide.md:19]: @martyav I edited your suggestion slightly, we can commit it.

File: docs/pages-for-subheaders/rke1-hardening-guide.md [hunk 5]
```diff
@@ -243,7 +246,6 @@ services:
   kubelet:
     extra_args:
       feature-gates: RotateKubeletServerCertificate=true
-      protect-kernel-defaults: "true"
```
Comment (macedogm) [inline docs/pages-for-subheaders/rke1-hardening-guide.md:246]: This needs to be inside a specific tab depending on the CIS benchmark version, because it's still needed in CIS 1.23 and CIS 1.24.

File: docs/pages-for-subheaders/rke1-hardening-guide.md [hunk 6]
```diff
@@ -293,7 +295,6 @@ services:
   kubelet:
     extra_args:
       feature-gates: RotateKubeletServerCertificate=true
-      protect-kernel-defaults: true
```
Comment (macedogm) [inline docs/pages-for-subheaders/rke1-hardening-guide.md:296]: Same as before:

> 

This needs to be inside a specific tab depending on the CIS benchmark version, because it's still needed in CIS 1.23 and CIS 1.24.

File: docs/pages-for-subheaders/rke2-hardening-guide.md [hunk 7]
```diff
@@ -64,22 +65,6 @@ sudo useradd -r -c "etcd user" -s /sbin/nologin -M etcd -U
 
 The runtime requirements to pass the CIS Benchmark are centered around pod security, network policies and kernel parameters. Most of this is automatically handled by RKE2 when using a valid `cis-1.xx` profile, but some additional operator intervention is required. These are outlined in this section.
 
-### Ensure `protect-kernel-defaults` is set
```
Comment (macedogm) [inline docs/pages-for-subheaders/rke2-hardening-guide.md:67]: Same as before:

> 

This needs to be inside a specific tab depending on the CIS benchmark version, because it's still needed in CIS 1.23 and CIS 1.24.

File: docs/pages-for-subheaders/rke2-hardening-guide.md [hunk 8]
```diff
@@ -89,15 +74,15 @@ RKE2 always runs with some amount of pod security.
...
-Below is the minimum necessary configuration needed for hardening RKE2 to pass CIS v1.23 hardened profile `rke2-cis-1.23-hardened` available in Rancher.
+Below is the minimum necessary configuration needed for hardening RKE2 to pass CIS v1.23 hardened profile `rke2-cis-1.7-hardened` available in Rancher.
 
 ```yaml
 spec:
   defaultPodSecurityAdmissionConfigurationTemplateName: rancher-restricted
   rkeConfig:
     machineSelectorConfig:
       - config:
-          profile: cis-1.23
```
Comment (macedogm) [inline docs/pages-for-subheaders/rke2-hardening-guide.md:100]: Since the new CIS flag isn't yet implemented in RKE2, we must advise customers to use only the `cis-1.23` flag for the moment.

File: docs/pages-for-subheaders/rke2-hardening-guide.md [hunk 9]
```diff
@@ -229,7 +214,7 @@ spec:
...
```
Comment (macedogm) [inline docs/pages-for-subheaders/rke2-hardening-guide.md:232]: Same as before:

> Since the new CIS flag isn't yet implemented in RKE2, we must advise customers to use only the `cis-1.23` flag for the moment.

**Review by macedogm — CHANGES_REQUESTED** [review]

**Review by macedogm — CHANGES_REQUESTED** [review]
@andypitcher can you please remove the RKE1 file `docs/pages-for-subheaders/rke1-hardening-guide.md` from this PR? I believe it's already being updated in its own PR https://github.com/rancher/rancher-docs/pull/794, no?

Otherwise the RKE2 file LGTM.

**Comment by andypitcher** [comment]
@macedogm RKE1 file `docs/pages-for-subheaders/rke1-hardening-guide.md` has been removed from this PR, through this [commit](https://github.com/rancher/rancher-docs/commit/7ae3799d6a595bb50350b45b0e3518ac38c4a668).

**Review by macedogm — APPROVED** [review]
